horovod/run/common/util/safe_shell_exec.py

```diff
@@ -150,22 +184,22 @@ def kill_executor_children_if_parent_dies():
     stdout_fwd = in_thread(target=forward_stream, args=(stdout_r, stdout, 'stdout', index))
     stderr_fwd = in_thread(target=forward_stream, args=(stderr_r, stderr, 'stderr', index))
 
-    # TODO: Currently this requires explicitly declaration of the event and signal handler to set
+    # TODO: Currently this requires explicitly declaration of the events and signal handler to set
     #  the event (gloo_run.py:_launch_jobs()). Need to figure out a generalized way to hide this behind
     #  interfaces.
-    stop = None
-    if event is not None:
-        stop = threading.Event()
-        on_event(event, os.kill, (middleman_pid, signal.SIGTERM), stop=stop, silent=True)
+    stop = threading.Event()
```

Comment:
since `stop` is always set to an `Event` (was `None` before), the `if` clause in the `finally` below can be removed and `stop.set` can always be done. If there are no events, that is redundant but simplifies the code.